Update top nav for desktop - lang more visible

johnnyd-eth opened the pull request:
## What does this PR do and why?

- Make lang selector more visible by moving it out of the dotdotdot menu. 
- Move disconnect button to the wallet tooltip
- Use short text (e.g. EN) for languages on desktop

Other code changes:
- Moves non-antd styles out of CSS and into JSX `navStyles`

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/156908475-26d28209-5111-4580-84b3-82f5d7d53618.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

johnnyd-eth commented:
TODO: 
- theme selector on mobile
- Language selector opens on click of globe 
- Action hover on disconnect button

johnnyd-eth commented:
TODO:

- [ ] fix resources dropdown bug
- [ ] get spacing perfect between lang->theme->wallet (use <Space/>)

aeolianeth commented on src/components/Navbar/MobileCollapse.tsx:
```diff
@@ -60,7 +60,7 @@ export default function MobileCollapse() {
           {menu(() => setActiveKey(activeKey === 0 ? undefined : 0))}
           <div className="nav-subsection">
-            <NavLanguageSelector />
+            <NavLanguageSelector mobile={true} />
```
**suggestion**: i think this is equiv to:

```suggestion
            <NavLanguageSelector mobile />
```

aeolianeth commented on src/components/Navbar/NavLanguageSelector.tsx:
```diff
@@ -40,15 +52,35 @@ export default function NavLanguageSelector({
+      style={{ cursor: 'pointer' }}
+      onClick={e => {
+        e.stopPropagation()
+        setDropdownOpen(dropdownOpen === true ? false : true)
```
```suggestion
        setDropdownOpen(!dropdownOpen)
```

johnnyd-eth replied:
Lol yep

aeolianeth commented on src/components/Navbar/NavLanguageSelector.tsx:
```diff
@@ -40,15 +52,35 @@ export default function NavLanguageSelector({
-        value={Languages[currentSelectedLanguage]?.long ?? 'English'}
+        dropdownStyle={mobile ? {} : desktopDropdownStyle}
+        open={dropdownOpen}
+        value={selectHeader ?? 'EN'}
```
**suggestion** might be possible to source the default language from `i18n.ts`, not sure.

aeolianeth commented on src/components/Navbar/ThemePicker.tsx:
```diff
@@ -1,46 +1,99 @@
 
+  if (!mobile) {
+    return (
+      <div
```
**suggestion**: would probably make sense to have a mobile and desktop specific component, to reduce complexity of this template. Not big deal though

aeolianeth commented on src/styles/antd-overrides/collapse.scss:
```diff
@@ -69,33 +69,37 @@
-  }
+}
+
+.nav-dropdown-item {
```
**question**; does this have to be a class? The best approach is debatable, but inline styles in JSX are more commonly used in the codebase. WDYT?

johnnyd-eth replied:
Yep it's a good call. I've moved pretty much all non-antd styles out of the `css` files and into a `navStyles.ts` typescript file

aeolianeth reviewed (changes requested):
@johnnyd-eth LGTM, left some minor comments. Also curious what less x-axis margin looks like on the nav items. Seems like there might be too much space either side of the lang selector and theme now, not sure.

aeolianeth commented on src/constants/theme/colors/index.ts:
```diff
@@ -6,10 +6,12 @@ import { backgroundColors } from './background'
   text: textColors[themeOption],
   icon: iconColors[themeOption],
   stroke: strokeColors[themeOption],
+  boxShadow: boxShadow[themeOption],
```
nice

aeolianeth commented on src/styles/antd-overrides/button.scss:
```diff
@@ -132,3 +132,7 @@
   box-shadow: 0px 8px 12px 2px #000000;
 }
+
+.copyIcon:hover {
```
**suggestion**: This isn't technically an ant-d override (same with the feedback button styles above). Worth thinking about moving these to their own file so we get a clearer separation. All g for now

aeolianeth commented on src/components/Navbar/Mobile/ResourcesDropdownMobile.tsx:
```diff
@@ -0,0 +1,99 @@
+  window.addEventListener('click', () => setActiveKey(undefined), false)
+
+  return (
+    <div className="resources-dropdown">
```
**suggestion**: any chance we can use the antd `SubMenu` here instead? Would simplify this quite a bit I think.

https://ant.design/components/menu/

https://user-images.githubusercontent.com/94939382/158013586-183b160f-5371-4a49-9846-975e9fb3dac9.mp4

johnnyd-eth replied:
Yep good idea I've switched to antd `SubMenu` and added its styles to antd-overrides

aeolianeth commented on src/components/Navbar/Mobile/ResourcesDropdownMobile.tsx:
```diff
@@ -0,0 +1,99 @@
+  const iconSize = 12
+
+  // Close dropdown when clicking anywhere in the window
+  window.addEventListener('click', () => setActiveKey(undefined), false)
```
**issue** these event listeners will get added every time the dropdown is opened. This might lead to unexpected behavior.

One solution would be to have a hook - something like `useWindowClick`, that only adds the event listener _once_. Then, we can use that hook throughout these components where necessary.


https://user-images.githubusercontent.com/94939382/158013814-f506c483-5c96-43ed-be86-8cbaa7577581.mp4

johnnyd-eth replied:
Yep good pickup. I changed it to how @torvusbug suggested we handle a similar previous instance of this. Wdyt? 

![Screen Shot 2022-03-16 at 2 02 43 pm](https://user-images.githubusercontent.com/96150256/158514958-95e7fcf6-6f2c-43a9-abcb-8b76bda26bcc.png)

aeolianeth reviewed (changes requested):
@johnnyd-eth some comments!

johnnyd-eth commented on src/components/shared/EtherscanLink.tsx:
```diff
@@ -12,11 +12,21 @@ import ExternalLink from './ExternalLink'
   value: string | undefined
   type: 'tx' | 'address'
+  shortened?: boolean
+  showText?: boolean
```
Yep

johnnyd-eth commented on src/components/shared/EtherscanLink.tsx:
```diff
@@ -12,11 +12,21 @@ import ExternalLink from './ExternalLink'
 }: {
   value: string | undefined
   type: 'tx' | 'address'
+  shortened?: boolean
```
Yep I like it

johnnyd-eth commented on src/components/Navbar/Mobile/MobileCollapse.tsx:
```diff
@@ -13,34 +13,45 @@ import { NetworkContext } from 'contexts/networkContext'
 
 export default function MobileCollapse() {
-  const [activeKey, setActiveKey] = useState<0 | undefined>()
+  const [navOpen, setNavOpen] = useState<0 | undefined>()
```
Yep good idea

johnnyd-eth commented on src/constants/styles/boxShadow.ts:
```diff
@@ -0,0 +1,3 @@
+export default function juiceBoxShadow(isDarkMode: boolean) {
+  return `0px 8px 12px ${isDarkMode ? '#000000' : 'rgba(0, 0, 0, 0.1)'}`
```
Just removed this whole thing, didn't need it

johnnyd-eth commented on src/components/Navbar/MenuItems.tsx:
```diff
@@ -83,27 +46,87 @@ export function DropdownItem({
+      direction={mobile ? 'vertical' : 'horizontal'}
+    >
+      {!mobile && (
+        <a href="/" style={{ display: 'inline-block' }}>
```
Navbar sits outside of Router so it errors. I think not worth rearranging. Wdyt?

aeolianeth replied:
Oh damn, TIL. No worries.

johnnyd-eth commented:
TODO:
- width of resources dropdown on mobile
- fix event listeners (dropdowns not closing on desktop when clicking outside it)

johnnyd-eth commented:
@aeolianeth addressed all your comments! LMKWYT

aeolianeth reviewed (approved):
@johnnyd-eth this LGTM! Thanks for addressing all the feedback. I added a few commits and am happy to approve!

One thing to keep in mind: make sure to check the browser console occasionally when devving. There were a few random Antd errors about Menu.Item that were introduced originally (fixed in https://github.com/jbx-protocol/juice-interface/pull/545/commits/4538ea278419d0c7feca2b4397cb1418569aa9f9 and https://github.com/jbx-protocol/juice-interface/pull/545/commits/4538ea278419d0c7feca2b4397cb1418569aa9f9).

johnnyd-eth commented:
> @johnnyd-eth this LGTM! Thanks for addressing all the feedback. I added a few commits and am happy to approve!
> 
> One thing to keep in mind: make sure to check the browser console occasionally when devving. There were a few random Antd errors about Menu.Item that were introduced originally (fixed in [4538ea2](https://github.com/jbx-protocol/juice-interface/commit/4538ea278419d0c7feca2b4397cb1418569aa9f9) and [4538ea2](https://github.com/jbx-protocol/juice-interface/commit/4538ea278419d0c7feca2b4397cb1418569aa9f9)).

Ahh yep ok nice pickup, and will definitely keep in mind for future. Let's merge this thang 🚀
